### PR TITLE
enh(host): Check for illegal characters when creating hosts

### PR DESCRIPTION
**Pull request by callapa**
## Description

Check for illegal characters when creating hosts.
Ckeck all mandatory properties.

**Fixes** # (issue)

Illegal characters were not deleted when saving a host requested by the autodiscovery module.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x
- [ ] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Domain/HostConfiguration/HostConfigurationService.php [hunk 1]
```diff
@@ -50,16 +62,49 @@ public function addHost(Host $host): int
...
+            if ($engineConfiguration === null) {
+                throw new HostConfigurationException(_('Impossible to find the Engine configuration'));
+            }
+
+            $safedHostName = EngineConfiguration::removeIllegalCharacters(
+                $host->getName(),
+                $engineConfiguration->getIllegalObjectNameCharacters()
+            );
+            if (empty($safedHostName)) {
+                throw new HostConfigurationException(_('Host name can not be empty'));
```
Comment (sc979) [inline src/Centreon/Domain/HostConfiguration/HostConfigurationService.php:89]:
```suggestion
                throw new HostConfigurationException(_('Host name cannot be empty'));
```

File: src/Centreon/Domain/MonitoringServer/MonitoringServerService.php [hunk 2]
```diff
@@ -74,6 +74,22 @@ public function findServer(int $monitoringServerId): ?MonitoringServer
...
+    /**
+     * @inheritDoc
+     */
+    public function findServerByName(string $monitoringServerName): ?MonitoringServer
+    {
+        try {
+            return $this->monitoringServerRepository->findServerByName($monitoringServerName);
+        } catch (\Exception $ex) {
+            throw new MonitoringServerException(
+                sprintf(_('Error when searching for a monitoring server %s'), $monitoringServerName),
```
Comment (sc979) [inline src/Centreon/Domain/MonitoringServer/MonitoringServerService.php:86]:
```suggestion
                sprintf(_('Error when searching monitoring server %s'), $monitoringServerName),
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -15532,3 +15532,6 @@ msgstr "Cet hôte n'est pas un modèle d'hôte"
 
 msgid "Error when searching for host and related service templates"
 msgstr "Erreur lors de la recherche d'hôtes et de modèles de service liés"
+
+msgid "Error when searching for a monitoring server %s"
```
Comment (sc979) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15536]:
```suggestion
msgid "Error when searching monitoring server %s"
```

File: src/Centreon/Domain/Engine/EngineService.php [hunk 4]
```diff
@@ -283,6 +283,18 @@ public function findEngineConfigurationByHost(\Centreon\Domain\HostConfiguration
...
 
+    /**
+     * @inheritDoc
+     */
+    public function findEngineConfigurationByName(string $engineName): ?EngineConfiguration
+    {
+        try {
+            return $this->engineConfigurationRepository->findEngineConfigurationByName($engineName);
+        } catch (\Throwable $ex) {
+            throw new EngineException(_('Error when searching for the Engine configuration'), 0, $ex);
```
Comment (sc979) [inline src/Centreon/Domain/Engine/EngineService.php:294]:
```suggestion
            throw new EngineException(_('Error when searching the Engine configuration'), 0, $ex);
```

Comment (sc979) [reply src/Centreon/Domain/Engine/EngineService.php:294]: Is this sentence translated too ?

File: src/Centreon/Domain/HostConfiguration/HostConfigurationService.php [hunk 5]
```diff
@@ -50,16 +62,49 @@ public function addHost(Host $host): int
             throw new HostConfigurationException(_('Host name can not be empty'));
         }
         try {
+            if (empty($host->getIpAddress())) {
+                throw new HostConfigurationException(_('Ip address can not be empty'));
```
Comment (sc979) [inline src/Centreon/Domain/HostConfiguration/HostConfigurationService.php:66]:
```suggestion
                throw new HostConfigurationException(_('Ip address cannot be empty'));
```

File: src/Centreon/Domain/HostConfiguration/HostConfigurationService.php [hunk 6]
```diff
@@ -50,16 +62,49 @@ public function addHost(Host $host): int
             throw new HostConfigurationException(_('Host name can not be empty'));
```
Comment (sc979) [inline src/Centreon/Domain/HostConfiguration/HostConfigurationService.php:62]:
```suggestion
            throw new HostConfigurationException(_('Host name cannot be empty'));
```

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**